python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+"""
+Kriging: metamodel with continuous and categorical variables
+============================================================
+"""
+# %%
+# We consider here the surrogate modeling of an analytical function characterized by
+# continuous and categorical variables
+#
+
+# %%
+import openturns as ot
+import openturns.experimental as otexp
+import numpy as np
+import matplotlib.pyplot as plt
+
+ot.RandomGenerator.SetSeed(5)
+
+# %%
+# We first show the advantage of modeling the various levels of a mixed
+# continuous / categorical function through a single surrogate model
+# on a simple test-case, defined below.
+
+
+# %%
+def illustrativeFunc(inp):
+    x, z = inp
+    x = 7 * x
+    y = np.cos(x) + 0.5 * z
+
+    return [y]
+
+
+dim = 2
+fun = ot.PythonFunction(dim, 1, illustrativeFunc)
+lvls = 2  # Number of discrete levels for z
+# Input distribution
+dist = ot.ComposedDistribution(
+    [ot.Uniform(0, 1), ot.UserDefined(np.arange(0, lvls).reshape((-1, 1)))]
```

Comment:
This is an example in OT, not NP:
```suggestion
    [ot.Uniform(0, 1), ot.UserDefined(ot.Sample.BuildFromPoint(range(lvls)))]
```